File: src/Gastromio.Persistence.MongoDB/OrderedDishInfoModel.cs
using System;

namespace Gastromio.Persistence.MongoDB
{
    public class OrderedDishInfoModel
    {
        public Guid ItemId { get; set; }
        public Guid DishId { get; set; }
        public string DishName { get; set; }
        public Guid VariantId { get; set; }
        public string VariantName { get; set; }
        public double VariantPrice { get; set; }
        public int Count { get; set; }
        public string Remarks { get; set; }
    }
}

File: src/Gastromio.App/ClientApp/src/app/order/models/ordered-dish-info.model.ts
export class OrderedDishInfoModel {

  constructor(init?: Partial<OrderedDishInfoModel>) {
    if (init) {
      Object.assign(this, init);
    }
  }

  public itemId: string;

  public dishId: string;

  public dishName: string;

  public variantId: string;

  public variantName: string;

  public variantPrice: number;

  public count: number;

  public remarks: string;

  public clone(): OrderedDishInfoModel {
    return new OrderedDishInfoModel({
      itemId: this.itemId,
      dishId: this.dishId,
      dishName: this.dishName,
      variantId: this.variantId,
      variantName: this.variantName,
      variantPrice: this.variantPrice,
      count: this.count,
      remarks: this.remarks
    });
  }

}
